### PR TITLE
Version 0.1 -- Initial implementation of Consul service discovery via DNS

### PR DESCRIPTION
**Pull request by cberube**

### CODE REVIEW

File: src/Consul/ConsulDns.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public function getServiceAddress($serviceName, $version = null)
+    {
+        $dnsEntry = "$serviceName.service.consul";
+
+        if (!empty($version)) {
+            $dnsEntry = "$version.$dnsEntry";
+        }
+
+        $resultList = call_user_func($this->lookupService, $dnsEntry, DNS_SRV + DNS_A);
```
Comment (nackjicholson) [inline src/Consul/ConsulDns.php]: Is `$this->lookupService` a callable or a string?

What does this function you pass in do? Looks like it has to return some 'ip' and 'port' info somehow.

Comment (cberube) [inline src/Consul/ConsulDns.php]: It is a callable, which is sometimes just a string (for built-in PHP functions).

`get_dns_record` queries DNS for a given entry and returns some or all records associated with the entry. In this case I get the SRV and A records -- the A record provides the IP address of the service and the SRV record provides the port number.

File: src/Consul/ConsulDns.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        if (is_null($lookupService)) {
+            $this->lookupService = "dns_get_record";
+        }
+    }
+
+    public function getServiceAddress($serviceName, $version = null)
+    {
+        $dnsEntry = "$serviceName.service.consul";
+
+        if (!empty($version)) {
```
Comment (notxt) [inline src/Consul/ConsulDns.php]: should `$version` be required? would you get all version of a service back if you don't provide it?

**Comment by nackjicholson** [comment]
You ready for this to go? :ok: :shipit:

**Comment by cberube** [comment]
Out of date and no longer valuable

**Pull request closed without merging**